test/Staking.test.js

```diff
@@ -0,0 +1,299 @@
+const Staking = artifacts.require("./Staking");
+const MyToken = artifacts.require("./MyToken");
+
+const {
+  ether,           // Big Number support
+  constants,    // Common constants, like the zero address and largest integers
+  expectEvent,  // Assertions for emitted events
+  expectRevert,
+  balance,
+  time, // Assertions for transactions that should fail
+} = require('@openzeppelin/test-helpers');
+
+const { expect } = require("chai");
+const BN = Web3.utils.BN;
+
+contract("Staking", async ([owner, acc2, acc3, acc4, acc5, acc6]) => {
+
+  let instanceToken;
+  let instanceStaking;
+
+  before(async () => {
+    instanceToken = await MyToken.new();
+    instanceStaking = await Staking.new(instanceToken.address);
+  });
+
+  describe("Deploy Vesting contract - false", async () => {
+    it("Error : Incorrect address , only contract address", async () => {
+      await expectRevert(Staking.new(acc2), "Error : Incorrect address , only contract address");
+    });
+  });
```

Comment:
should add tests for check corectly initialize stakingParams, and token address after deploy